src/L6/datasorter.py

```diff
@@ -18,7 +20,23 @@ def __init__(self):
 
     # Exercise 25
     def set_input_data(self, file_path_name: str):
-        pass
+        if type(file_path_name) is not str:
+            raise TypeError
+        file = Path(file_path_name)
+        if not file.is_file():
```

Comment:
I recommend declaring `import os` and use
```suggestion
        if os.path.exists(file_path_name) and os.path.isfile(file_path_name):
```
The os.path module supports some edge cases. More info here https://docs.python.org/3/library/os.path.html#module-os.pathhttps://docs.python.org/3/library/os.path.html#module-os.path